exist-core/src/main/java/org/exist/xquery/functions/fn/FunMatches.java

```diff
@@ -219,19 +216,15 @@ public NodeSet preSelect(Sequence contextSequence, boolean useContext) throws XP
         final boolean caseSensitive = !hasCaseInsensitive(flags);
 
         final String pattern;
-		if(isCalledAs("matches-regex")) {
```

Comment:
Why is `isCalledAs("matches-regex")`  removed from the code?
Is it no longer necessary, was it previously deprecated?

Reply:
The function definition was removed sometime ago, but it seems that we forgot to remove some of the implementation.